include/matx/core/tensor.h

```diff
@@ -877,7 +878,7 @@ class tensor_t : public detail::tensor_impl_t<T,RANK,Desc> {
   {
     MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
 
-    static_assert(RANK >= 2, "Only tensors of rank 2 and higher can be permuted.");
+    static_assert(RANK >= 1, "Only tensors of rank 1 and higher can be permuted.");
```

Comment:
What's the reason to permute a 1D tensor?

Reply:
We have to take the max along the last dimension of the singular values vector, and max() calls permute. The vector could be of rank 1 if it isn’t batched.